Add abilitiy to configure different git hosting services.

cranst0n opened the pull request:
First run at #123. Note that I also upgraded FontAwesome to 4.7.0 only because it added a GitLab icon in 4.6.0.

raulraja commented on src/main/scala/microsites/MicrositeKeys.scala:
```diff
@@ -54,5 +54,9 @@ trait MicrositeKeys {
   val micrositeGithubRepo  = settingKey[String]("Microsite Github repo")
+  val micrositeGitHostingService =
+    settingKey[String]("Service used for git hosting. By default, it'll be 'Github'.")
+  val micrositeGitHostingUrl = settingKey[String](
```
can we set this automatically to empty strings when the user specifies a different `micrositeGitHostingService` automatically so they don't have to do that themselves?

cranst0n commented:
@raulraja Updated with your recommendations. Also added some JS that will hide the github related watching/stars stats if it's not a GitHub project.

cranst0n commented:
Is Travis is failing because it's a chicken and egg situation or is it legit? This is my first dive into an SBT plugin. It passed locally after a `publish-local`.

raulraja commented:
Looks like the scripted tests are failing. They need to be adjusted to reflect the new settings.  We also need to make sure this change does not break existing configurations for other sites. Is any of the new keys or settings mandatory or change the way the setup works? Also instead of having the stats and stuff from github.com hidden perhaps we should not make the call to github.com unless that is the provider applied to a given configuration. Potentially we could add a new setting key to not fetch github.com stats and contributions which would be useful regardless for people not wanting to show those.

cranst0n commented:
Right. I'm just not sure if the test I added is only failing because it's trying to fetch `0.4.1-SHAPSHOT` and coming back with the last snapshot that was published (which would not include any of the setting keys I've added) or if there's something else I'm missing/understanding. After I did a `publish-local` on my local machine, all tests passed. It looks as though my test is the only one failing on Travis.

These new settings should be optional and have no effect if they're not used.

I will update to undo hiding the GitHub stat indicators.

cranst0n commented:
@raulraja Any guidance on how you want me to proceed with the test?

raulraja commented:
@juanpedromoreno is there anything special that needs to happen for the scripted test to work in this scenario?

juanpedromoreno commented:
@cranst0n @raulraja I've just merged this PR https://github.com/47deg/sbt-microsites/pull/126. 

Could you try again rebasing from upstream master? Thanks!

raulraja commented:
@cranst0n thanks for your contribution!

cranst0n commented:
@raulraja @juanpedromoreno Thanks for working to get this in. Happy to make a small contribution to a really great project!
